File: reactive-web/src/main/java/com/github/corneil/reactivespring5/service/LocationHistoryInterface.java
package com.github.corneil.reactivespring5.service;

import com.github.corneil.model.LocationHistory;
import com.github.corneil.reactivespring5.model.ExtendedLocationHistory;
import reactor.core.publisher.Flux;

import java.util.Date;

public interface LocationHistoryInterface {
	Flux<LocationHistory> findByDates(Date startDate, Date endDate);
	Flux<ExtendedLocationHistory> findAndConvert(Date startDate, Date endDate);
}

File: reactive-kotlin/src/main/kotlin/com/github/corneil/reactivespring5kotlin/service/LocationHistoryInterface.kt
package com.github.corneil.reactivespring5kotlin.service

import com.github.corneil.reactivespring5kotlin.model.ExtendedLocationHistory
import com.github.corneil.reactivespring5kotlin.model.LocationHistory
import reactor.core.publisher.Flux

import java.util.Date

interface LocationHistoryInterface {
    fun findByDates(startDate: Date, endDate: Date): Flux<LocationHistory>
    fun findAndConvert(startDate: Date, endDate: Date): Flux<ExtendedLocationHistory>
}
